amqtt/broker.py

```diff
@@ -673,6 +673,13 @@ async def _handle_message_delivery(
         self.logger.debug(f"{client_session.client_id} handling message delivery")
         app_message = wait_deliver.result()
 
+        # notify of a message's receipt, even if a client isn't necessarily allowed to send it
```

Comment:
in terms of nomenclature (and functionality), i think we should differentiate the event of 'receiving' a message from whether a message is going to be 'broadcast'. the difference in the two is that a message received might be on a topic where that client is not allowed to publish (either from topic filtering or other mqtt restrictions).

this allows a plugin -- such as this shadow -- to receive messages on `$` topics, even the client is not allowed by spec to have these messages published directly. it could also be used in cases where a plugin wants to monitor 'bad actors' who might be continuously publishing messages to wrong topics or a plugin which can observe what messages are being blocked and see if there is an incorrect permission preventing a client from sending a message.